MechJeb2/PartExtensions.cs

```diff
@@ -102,31 +102,17 @@ public static bool IsSepratron(this Part p)
 
         public static bool IsEngine(this Part p)
         {
-            for (int i = 0; i < p.Modules.Count; i++)
-            {
-                PartModule m = p.Modules[i];
-                if (m is ModuleEngines) return true;
-            }
-            return false;
+            return p.FindModuleImplementing<ModuleEngines>() != null;

```

Comment:
Unity Profiler
+
KSP in dev mode ( http://forum.kerbalspaceprogram.com/index.php?/topic/102909-ksp-plugin-debugging-and-profiling-for-visual-studio-and-monodevelop-on-all-os/ ) 
+ 
pairs of Profiler.BeginSample / Profiler.EndSample
Like there 
https://github.com/MuMech/MechJeb2/blob/master/MechJeb2/DisplayModule.cs#L164-L169
+
MJ compiled with the ENABLE_PROFILER symbol (Profiler.BeginSample uses a Conditional attribute)